### PR TITLE
[Script] Fix broken tests after conflicting merge

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
culprits are #16920 and #16893

In this PR, we split properly Stable tests from Dev one.

### CODE REVIEW

File: daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DamlScriptTestRunner.scala [hunk 1]
```diff
@@ -58,6 +58,14 @@ class DamlScriptTestRunner extends AnyWordSpec with CantonFixture with Matchers
...
           |ScriptTest:tree SUCCESS
           |ScriptTest:tupleKey SUCCESS
+          |TestContractId:testContractId SUCCESS
+          |TestExceptions:test SUCCESS
+          |TestExceptions:try_catch_recover SUCCESS
+          |TestExceptions:try_catch_then_abort FAILURE (com.daml.lf.engine.script.Runner$InterpretationError: Error: Unhandled Daml exception: DA.Exception.GeneralError:GeneralError@XXXXXXXX{ message = "expected exception" })
+          |TestExceptions:try_catch_then_error FAILURE (com.daml.lf.engine.script.Runner$InterpretationError: Error: Unhandled Daml exception: DA.Exception.GeneralError:GeneralError@XXXXXXXX{ message = "expected exception" })
+          |TestExceptions:try_catch_then_fail FAILURE (com.daml.lf.engine.script.Runner$InterpretationError: Error: Unhandled Daml exception: DA.Exception.GeneralError:GeneralError@XXXXXXXX{ message = "expected exception" })
+          |TestInterfaces:test SUCCESS
+          |TestInterfaces:test_queryInterface SUCCESS
```
Comment (remyhaemmerle-da) [inline daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DamlScriptTestRunner.scala:68]: Interface and exception tests are now stable

File: daml-script/test/BUILD.bazel [hunk 2]
```diff
@@ -27,6 +27,9 @@ genrule(
       mkdir -p $$TMP_DIR/daml
       cp -L $(location :daml/ScriptTest.daml) $$TMP_DIR/daml
       cp -L $(location :daml/MultiTest.daml) $$TMP_DIR/daml
+      cp -L $(location :daml/TestContractId.daml) $$TMP_DIR/daml
+      cp -L $(location :daml/TestExceptions.daml) $$TMP_DIR/daml
+      cp -L $(location :daml/TestInterfaces.daml) $$TMP_DIR/daml
```
Comment (remyhaemmerle-da) [inline daml-script/test/BUILD.bazel:32]: Those test stable features.

File: daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DevIT.scala [hunk 3]
```diff
@@ -13,33 +13,33 @@ import org.scalatest.Inside
...
-    with Matchers {
+final class DevIT extends AsyncWordSpec with AbstractScriptTest with Inside with Matchers {
   final override protected lazy val devMode = true
   final override protected lazy val timeMode = ScriptTimeMode.WallClock
 
-  val coinV1DarPath = BazelRunfiles.rlocation(Paths.get("daml-script/test/coin-v1.dar"))
-  val coinV2DarPath = BazelRunfiles.rlocation(Paths.get("daml-script/test/coin-v2.dar"))
-  val coinV2NewFieldDarPath =
+  lazy val devDarPath = BazelRunfiles.rlocation(Paths.get("daml-script/test/script-test-1.dev.dar"))
+  lazy val devDar = CompiledDar.read(devDarPath, Runner.compilerConfig)
```
Comment (remyhaemmerle-da) [inline daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/DevIT.scala:21]: we group upgrade and choice authority test in one unique Dev test.

**Review by cocreature — APPROVED** [review]
thx

**Pull request merged**